### PR TITLE
Added option for retirement rarity

### PR DESCRIPTION
**Pull request by wiseasswolfofyoitsu**
* Broke out Retirement module into its own section in the config
* Added the ability to turn off retiring rares.

### CODE REVIEW

**Comment by Egoistically** [comment]
Can you also add an option to disable commons? Doing so makes no sense but just to have the option available for any possible psychopaths.

**Comment by wiseasswolfofyoitsu** [comment]
Would be a pretty trivial mod, yeah.

**Comment by wiseasswolfofyoitsu** [comment]
Ok, option for commons retiring added, along with some config safety checks, for all the pedants out there!

File: util/config.py [hunk 1]
```diff
@@ -57,12 +57,18 @@ def read(self):
...
             # Legacy retirement setting. Maintain classic rare retirement behavior
             self.retirement['enabled'] = config.getboolean('Modules', 'Retirement')
             self.retirement['rares'] = True
+            self.retirement['commons'] = True
         else:
             self.retirement['enabled'] = False
+            
+        if self.retirement['enabled'] and not (self.retirement['commons'] or self.retirement['rares']):
+            Logger.log_error("Invalid config. Retirement is enabled, but no ship rarities are selected.")
+            sys.exit(1)
```
Comment (Egoistically) [inline util/config.py]: This should go inside `validate` function (around line 199), and instead of using `sys.exit(1)` you should use `self.ok = False`.

**Review by Egoistically — CHANGES_REQUESTED** [review]

**Comment by wiseasswolfofyoitsu** [comment]
Change request should be resolved

**Review by Egoistically — APPROVED** [review]

**Comment by Egoistically** [comment]
I have another question, doesn't this `if not (self.retirement['commons'] or self.retirement['rares'])` mean that if either of both is disabled it won't work?

**Comment by wiseasswolfofyoitsu** [comment]
It only goes into the error handler if both are disabled and passes if either or both are enabled. It's logically equivalent to `if Not Commons and Not Rares`

**Comment by Egoistically** [comment]
Aight, looks good to me.

**Pull request merged**